tools/system_libs.py

```diff
@@ -877,11 +879,12 @@ def get_files(self):
           'pthread_self_stub.c'
         ])
 
-    # These are included in wasm_libc_rt instead
-    ignore += [os.path.basename(f) for f in get_libc_rt_files(superset=True)]
+    if self.is_optz:
+      ignore += ['pow.c', 'pow_data.c', 'log.c', 'log_data.c', 'log2.c', 'log2_data.c']
+    else:
+      ignore += ['pow_small.c', 'log_small.c', 'log2_small.c']
```

Comment:
I'm not sure I see how this PR makes things simpler. Now we need this logic here, and it is separate from the main listing of these files - where in that main listing we have duplicate symbols, so that code is only valid with this ignore-list basically. That seems a little confusing expecially since the code is far apart.

Also, I'm not sure yet how this works, but it looks like this adds a new variation to all of libc for `-Oz`? Are we sure that's what we want? Aside from variations to the largest library being somewhat costly, this may make some things noticeably slower than before. Perhaps we should measure that?